src/XMLGenerator.php

```diff
@@ -357,10 +357,15 @@ public function updateUser(
                 );
             }
         }
+
+        // If the email address is being changed, or if the SendEmailTo field
+        // is set to 'Self', then the new email address must be included in
+        // the request.
         $info = $userTag->addChild('Info');
-        if (!empty($user->getOldEmail())) {
+        if (!empty($user->getOldEmail()) || $user->getSendEmailTo() == 'Self') {
             $info->addChild('Email', $user->getEmail());
         }
+        
         if (!empty($user->getOldEmployeeId())) {
```

Comment:
There is an issue in these lines; as written we can only change the user's email address when identifying them by email address and we can only change their `Employee Id` only when identifying them by employee id. On top of that you encountered the issue that SmarterU's API requires we resubmit the user's email address when setting the `Send Email To` property to `self` because you were trying to send a slightly mutated user read from SmarterU back to SmarterU. Fixing the first issue though I think will also fix the second in a cleaner way.

```suggestion
        if (!empty($user->getEmail()) {
            $info->addChild('Email', $user->getEmail());
        }
        
        if (!empty($user->getEmployeeId())) {
```